Core/clim-basic/text-formatting.lisp

```diff
@@ -198,12 +198,15 @@
 (defgeneric invoke-with-indenting-output
     (stream continuation &key indent move-cursor)
   (:method (stream continuation &key indent (move-cursor t))
-    (let ((left-margin (copy-list (getf (stream-text-margins stream) :left))))
+    (let ((left-margin (copy-list (getf (stream-text-margins stream) :left)))
+          (line-beginning (page-cursor-initial-position stream)))
       (setf (second left-margin)
             (+ (parse-space stream (second left-margin) :horizontal)
                (parse-space stream indent :horizontal)))
       (with-temporary-margins (stream :left left-margin :move-cursor move-cursor)
-        (funcall continuation stream)))))
+        (unwind-protect (funcall continuation stream)
+          (when (stream-start-line-p stream)
+            (setf (slot-value (stream-text-cursor stream) 'x) line-beginning)))))))
```

Comment:
Can this be done using accessors?

Reply:
it could with a considerable effort to disable protocols 

```
(when (stream-start-line-p stream)
(let ((cursor (stream-text-cursor stream)))
  (with-cursor-off (cursor)
     (setf (cursor-position cursor) (values line-beginning
                                                                     (nth-value 1 (cursor-position cursor)))))))
```

but* it is precisely done this way to bypass accessor protocols (it is a concious choice after writing a code similar to the above). do you want me to change it?

Reply:
Sounds like keeping the code as it is but adding a comment would be best.

Reply:
done